hack/e2e.go

```diff
@@ -292,18 +293,17 @@ func run(deploy deployer) error {
 			return fmt.Errorf("starting e2e cluster: %s", err)
 		}
 		if *dump != "" {
-			cmd := exec.Command("./cluster/kubectl.sh", "--match-server-version=false", "get", "nodes", "-oyaml")
-			b, err := cmd.CombinedOutput()
-			if *verbose {
-				log.Printf("kubectl get nodes:\n%s", string(b))
-			}
-			if err == nil {
-				if err := ioutil.WriteFile(filepath.Join(*dump, "nodes.yaml"), b, 0644); err != nil {
-					errs = appendError(errs, fmt.Errorf("error writing nodes.yaml: %v", err))
+			errs = appendError(errs, xmlWrap("get nodes", func() error {
```

Comment:
To make this easier to read, could you make this anonymous function a named one defined elsewhere on its own and just call it here?